firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt

```diff
@@ -138,9 +138,14 @@ internal constructor(
       return
     }
 
-    sessionCoordinator.attemptLoggingSessionEvent(
-      SessionEvents.startSession(firebaseApp, sessionDetails, sessionSettings, subscribers)
-    )
+    try {
+      val sessionEvent =
+        SessionEvents.startSession(firebaseApp, sessionDetails, sessionSettings, subscribers)
+      sessionCoordinator.attemptLoggingSessionEvent(sessionEvent)
+    } catch (ex: IllegalStateException) {
+      // This can happen if the app suddenly deletes the instance of FirebaseApp.
+      Log.i(TAG, "Sessions SDK failed to log Session Start event.", ex)
```

Comment:
Let's make this log message a bit more verbose as well. Basically, we want to signal to the developer that unregistering a FirebaseApp would not collect any data.